### PR TITLE
expression: implement vectorized evaluation for `builtinJSONLengthSig`

### PR DESCRIPTION
**Pull request by js00070**
### What problem does this PR solve? 
implement vectorized evaluation for `builtinJSONLengthSig` at https://github.com/pingcap/tidb/issues/12104

### What is changed and how it works?
the performance is related to the test case. It will get faster from 1% to 6 times. 
```
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-VecBuiltinFunc-12                 300879              3545 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-NonVecBuiltinFunc-12               67075             17767 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-VecBuiltinFunc#01-12                2026            525188 ns/op          301371 B/op       2410 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-NonVecBuiltinFunc#01-12             2137            554590 ns/op          301741 B/op       2410 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-VecBuiltinFunc#02-12                2059            519725 ns/op          302338 B/op       2422 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONLengthSig-NonVecBuiltinFunc#02-12             2120            540376 ns/op          302575 B/op       2422 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_json_vec.go [hunk 1]
```diff
@@ -245,11 +245,79 @@ func (b *builtinJSONKeys2ArgsSig) vecEvalJSON(input *chunk.Chunk, result *chunk.
...
+			return err
+		}
+		result.MergeNulls(jsonBuf, pathBuf)
+		for i := 0; i < nr; i++ {
+			if result.IsNull(i) {
+				continue
+			}
+			jsonItem := jsonBuf.GetJSON(i)
+
+			if jsonItem.TypeCode != json.TypeCodeObject && jsonItem.TypeCode != json.TypeCodeArray {
```
Comment (XuHuaiyu) [inline expression/builtin_json_vec.go]: Why change the execution logic here? The order will affect the returned error.

Comment (js00070) [reply expression/builtin_json_vec.go]: I have fixed the order problem. Now the logic is the same as the non-vec version.

File: expression/builtin_json_vec.go [hunk 2]
```diff
@@ -245,11 +245,85 @@ func (b *builtinJSONKeys2ArgsSig) vecEvalJSON(input *chunk.Chunk, result *chunk.
...
+			if err != nil {
+				return err
+			}
+			if pathExpr.ContainsAnyAsterisk() {
+				return json.ErrInvalidJSONPathWildcard
+			}
+
+			obj, exists := jsonItem.Extract([]json.PathExpression{pathExpr})
+			if !exists {
+				result.SetNull(i, true)
```
Comment (Reminiscent) [inline expression/builtin_json_vec.go]: We should do ```continue``` after this line

Comment (js00070) [reply expression/builtin_json_vec.go]: yes, we should. I forgot it.

**Comment by XuHuaiyu** [comment]
@js00070 Please resolve the conflicts.

**Comment by qw4990** [comment]
@js00070 Friendly ping, please solve conflicts.

**Comment by js00070** [comment]
> @js00070 Friendly ping, please solve conflicts.

conflicts have been solved now.

**Pull request merged**